File: Notepad#.java
#https://codeforces.com/contest/1766/problem/B
import java.util.*;
public class Notepad_Sharp {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        for(int t=sc.nextInt();t>0;t--) {
            int n=sc.nextInt(),c=0;
            String s=sc.next();
            boolean dejavu=false;
            Map<String,Boolean> m=new HashMap<>();
            for(int i=1;i<n;i++) {
                String sT="";
                sT+=s.charAt(i-1);
                sT+=s.charAt(i);
                if(s.charAt(i)==s.charAt(i-1)) c++;
                else c=0;
                if(m.getOrDefault(sT,false) && c!=2){
                    dejavu=true;
                    break;
                } 
                m.put(sT,true);
            }
            if(dejavu) System.out.println("YES");
            else System.out.println("NO");
        }
    }
}

File: Notepad#.py
#https://codeforces.com/contest/1766/problem/B
t=int(input())
for x in range(t):
    n=int(input())
    s=input()
    c=0
    dejavu=False
    dict={}
    i=1
    while i<n:
        sT=s[i-1]+s[i]
        if s[i]==s[i-1]:
            c+=1
        else:
            c=0
        if sT in dict and c!=2:
            dejavu=True
            break
        dict[sT]=True
        i+=1
    if dejavu:
        print("YES")
    else:
        print("NO")